### ISSUE TITLE
`rvm list remote` not showing binaries for Ubuntu 14.04 box

### ISSUE THREAD

**Issue by grahamb** [rvm/rvm#3180]
I'm trying to provision a dev box with rvm on Ubuntu 14.04 (via Vagrant and Ansible). I've installed rvm:

```
gpg --keyserver hkp://keys.gnupg.net --recv-keys D39DC0E3
\curl -sSL https://get.rvm.io | bash -s stable
```

When I try to `rvm install 1.9.3` or `rvm install ruby-1.9.3-p551 it says that no binaries can be found, and it'll compile from source.

```
$ rvm install ruby-1.9.3-p551
Searching for binary rubies, this might take some time.
No binary rubies available for: ubuntu/14.04/x86_64/ruby-1.9.3-p551.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
```

There _are_ binaries available, though: https://github.com/wayneeseguin/rvm/blob/master/config/remote & https://rvm_io.global.ssl.fastly.net/binaries/ubuntu/14.04/x86_64/

Doing `rvm --debug list remote` yields:

```
__list_remote_all https://rvm_io.global.ssl.fastly.net/binaries ubuntu/14.04/x86_64
__list_remote_all1 https://s3.amazonaws.com/jruby.org downloads
__list_remote_for_s3-0 https://s3.amazonaws.com/jruby.org?prefix=downloads/
Running(15): curl --fail --location --max-redirs 10 --max-time 1800 --connect-timeout 30 --retry-delay 2 --retry 3 -s https://s3.amazonaws.com/jruby.org?prefix=downloads/
__list_remote_for_s3-1 https://s3.amazonaws.com/jruby.org?prefix=downloads/&marker=downloads/1.6.5/jruby-bin-1.6.5.tar.gz.sha1
Running(15): curl --fail --location --max-redirs 10 --max-time 1800 --connect-timeout 30 --retry-delay 2 --retry 3 -s https://s3.amazonaws.com/jruby.org?prefix=downloads/&marker=downloads/1.6.5/jruby-bin-1.6.5.tar.gz.sha1
__list_remote_all2 http://binaries.rubini.us ubuntu/14.04/x86_64
__list_remote_for_s3-0 http://binaries.rubini.us?prefix=ubuntu/14.04/x86_64/
Running(15): curl --fail --location --max-redirs 10 --max-time 1800 --connect-timeout 30 --retry-delay 2 --retry 3 -s http://binaries.rubini.us?prefix=ubuntu/14.04/x86_64/
__list_remote_all3 https://rubies.travis-ci.org ubuntu/14.04/x86_64

# Remote rubies available:

   jruby-0.9.0
   jruby-1.6.6
   jruby-1.6.7
   jruby-1.6.7.2
   jruby-1.6.8
   jruby-1.7.0
   jruby-1.7.0.preview1
   jruby-1.7.0.preview2
   jruby-1.7.0.RC1
   jruby-1.7.0.RC2
   jruby-1.7.1
   jruby-1.7.10
   jruby-1.7.11
   jruby-1.7.12
   jruby-1.7.13
   jruby-1.7.14
   jruby-1.7.15
   jruby-1.7.16
   jruby-1.7.16.1
   jruby-1.7.2
   jruby-1.7.3
   jruby-1.7.4
   jruby-1.7.5
   jruby-1.7.6
   jruby-1.7.7
   jruby-1.7.8
   jruby-1.7.9
   rubinius-2.3.0

# * - installed already

# To download and install: rvm install {name} --binary

__rvm_rm_rf already gone: /home/vagrant/.rvm/tmp/18526*
```

Using `rvm mount` does work, but I'd like to avoid that as I'm trying to keep this vagrant box as generic as possible.

```
$ rvm mount -r https://rvm.io/binaries/ubuntu/14.04/x86_64/ruby-1.9.3-p551.tar.bz2
https://rvm.io/binaries/ubuntu/14.04/x86_64/ruby-1.9.3-p551.tar.bz2 - #configure
ruby-1.9.3-p551 - #download
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 13.3M  100 13.3M    0     0   343k      0  0:00:39  0:00:39 --:--:--  350k
ruby-1.9.3-p551 - #validate archive
ruby-1.9.3-p551 - #extract
ruby-1.9.3-p551 - #validate binary
ruby-1.9.3-p551 - #setup
ruby-1.9.3-p551 - #gemset created /home/vagrant/.rvm/gems/ruby-1.9.3-p551@global
ruby-1.9.3-p551 - #importing gemset /home/vagrant/.rvm/gemsets/global.gems....................................
ruby-1.9.3-p551 - #generating global wrappers........
ruby-1.9.3-p551 - #gemset created /home/vagrant/.rvm/gems/ruby-1.9.3-p551
ruby-1.9.3-p551 - #importing gemsetfile /home/vagrant/.rvm/gemsets/default.gems evaluated to empty gem list
ruby-1.9.3-p551 - #generating default wrappers......
```

I'm not quite sure what I'm doing wrong, but I'm sure I _am_ doing something wrong.

**Comment by grahamb** [comment]
I'm having a similar problem again. This time, the binaries show up in `rvm list remote` but fail to install:

```
$ rvm list remote

# Remote rubies available:

   jruby-0.9.0
   jruby-1.6.6
   jruby-1.6.7
   jruby-1.6.7.2
   jruby-1.6.8
   jruby-1.7.0
   jruby-1.7.0.preview1
   jruby-1.7.0.preview2
   jruby-1.7.0.RC1
   jruby-1.7.0.RC2
   jruby-1.7.1
   jruby-1.7.10
   jruby-1.7.11
   jruby-1.7.12
   jruby-1.7.13
   jruby-1.7.14
   jruby-1.7.15
   jruby-1.7.16
   jruby-1.7.16.1
   jruby-1.7.16.2
   jruby-1.7.17
   jruby-1.7.18
   jruby-1.7.2
   jruby-1.7.3
   jruby-1.7.4
   jruby-1.7.5
   jruby-1.7.6
   jruby-1.7.7
   jruby-1.7.8
   jruby-1.7.9
   rubinius-2.3.0
   ruby-1.9.3-p547
   ruby-1.9.3-p551
   ruby-2.0.0-p481
   ruby-2.0.0-p576
   ruby-2.0.0-p598
   ruby-2.1.2
   ruby-2.1.3
   ruby-2.1.5

# * - installed already

# To download and install: rvm install {name} --binary

$ rvm install ruby-1.9.3-p551 --binary
ruby-1.9.3-p551 - #removing src/ruby-1.9.3-p551..
Searching for binary rubies, this might take some time.
Requested binary installation but no rubies are available to download, consider skipping --binary flag.
```

I tried running `rvm get head` but that didn't change anything.

```
$ rvm get head
Downloading https://get.rvm.io
Downloading https://raw.githubusercontent.com/wayneeseguin/rvm/master/binscripts/rvm-installer.asc
Verifying /usr/local/lib/rvm/archives/rvm-installer.asc
gpg: Signature made Sun 11 Jan 2015 02:55:55 AM PST using RSA key ID BF04FF17
gpg: Good signature from "Michal Papis (RVM signing) <mpapis@gmail.com>"
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: 409B 6B17 96C2 7546 2A17  0311 3804 BB82 D39D C0E3
     Subkey fingerprint: 62C9 E5F4 DA30 0D94 AC36  166B E206 C29F BF04 FF17
GPG verified '/usr/local/lib/rvm/archives/rvm-installer'
Downloading https://github.com/wayneeseguin/rvm/archive/master.tar.gz

Upgrading the RVM installation in /usr/local/lib/rvm/
Upgrade of RVM in /usr/local/lib/rvm/ is complete.

# Administrator,
#
#   Thank you for using RVM!
#   We sincerely hope that RVM helps to make your life easier and more enjoyable!!!
#
# ~Wayne, Michal & team.

In case of problems: http://rvm.io/help and https://twitter.com/rvm_io

Upgrade Notes:

  * No new notes to display.

RVM reloaded!
$ rvm list remote

# Remote rubies available:

   jruby-0.9.0
   jruby-1.6.6
   jruby-1.6.7
   jruby-1.6.7.2
   jruby-1.6.8
   jruby-1.7.0
   jruby-1.7.0.preview1
   jruby-1.7.0.preview2
   jruby-1.7.0.RC1
   jruby-1.7.0.RC2
   jruby-1.7.1
   jruby-1.7.10
   jruby-1.7.11
   jruby-1.7.12
   jruby-1.7.13
   jruby-1.7.14
   jruby-1.7.15
   jruby-1.7.16
   jruby-1.7.16.1
   jruby-1.7.16.2
   jruby-1.7.17
   jruby-1.7.18
   jruby-1.7.2
   jruby-1.7.3
   jruby-1.7.4
   jruby-1.7.5
   jruby-1.7.6
   jruby-1.7.7
   jruby-1.7.8
   jruby-1.7.9
   rubinius-2.3.0
   ruby-1.9.3-p547
   ruby-1.9.3-p551
   ruby-2.0.0-p481
   ruby-2.0.0-p576
   ruby-2.0.0-p598
   ruby-2.1.2
   ruby-2.1.3
   ruby-2.1.5

# * - installed already

# To download and install: rvm install {name} --binary

$ rvm install ruby-1.9.3-p551 --binary
Searching for binary rubies, this might take some time.
Requested binary installation but no rubies are available to download, consider skipping --binary flag.
```

**Comment by mpapis** [comment]
not the same problem, this one could be network timeout issue, please try:
```bash
rvm install ruby-1.9.3-p551 --binary --max-time 30
```

**Comment by monfresh** [comment]
I'm seeing this issue as well when trying to install Ruby 2.1.5 on a Vagrant machine. It says that no binaries can be found, but I do see `ruby-2.1.5` when calling `rvm list remote`. I'm using RVM 1.26.10.

Here is the relevant portion in my script:
```bash
echo installing RVM
sudo -u vagrant -H bash -l -c 'gpg --keyserver hkp://keys.gnupg.net \
  --recv-keys D39DC0E3 && curl --silent -L https://get.rvm.io | bash -s stable'

source /home/vagrant/.rvm/scripts/rvm

echo installing Ruby 2.1.5
sudo -u vagrant -H bash -l -c '/home/vagrant/.rvm/bin/rvm install ruby-2.1.5 \
  --quiet-curl --autolibs=enabled && rvm alias create default 2.1.5'
```

**Comment by monfresh** [comment]
This worked though: `rvm install 2.1.5 --binary --max-time 30`

**Comment by pamci** [comment]
I am trying to install jruby from jenkins CI tool, i see the below error.no binaries found 

jruby-1.7.19 is not installed.
To install do: 'rvm install jruby-1.7.19'
Unknown ruby string (do not know how to handle): jruby-1.7.11.
jruby-1.7.11 is not installed - installing.
Unknown ruby string (do not know how to handle): jruby-1.7.11.
[32mjruby-1.7.11 - #removing src/jruby-1.7.11[0m|/-\|/-\|.-\.
Searching for binary rubies, this might take some time.
Unknown ruby string (do not know how to handle): jruby-1.7.11.
Unknown ruby string (do not know how to handle): jruby-1.7.11.
Unknown ruby string (do not know how to handle): jruby-1.7.11.
Unknown ruby string (do not know how to handle): jruby-1.7.11.
Unknown ruby string (do not know how to handle): jruby-1.7.11.
Unknown ruby string (do not know how to handle): jruby-1.7.11.
Unknown ruby string (do not know how to handle): jruby-1.7.11.
Unknown ruby string (do not know how to handle): jruby-1.7.11.
No binary rubies available for: redhat/6/x86_64/jruby-1.7.11.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for redhat.
Requirements installation successful.
Now using system ruby.
$JAVA_HOME was empty, setting up JAVA_HOME=/usr/lib/jvm/java-1.7.0-openjdk-1.7.0.65.x86_64, if it fails try setting JAVA_HOME to something sane and try again.
HEAD is now at 32f5af0 Bump for version
Previous HEAD position was 32f5af0... Bump for version
Switched to branch 'master'
From https://github.com/jruby/jruby
 * branch            master     -> FETCH_HEAD
Already up-to-date.
git checkout 1.7.11
Copying from repo to src path...
[32mjruby-1.7.11 - #mvn[0m|.
Error running 'mvn',
showing last 15 lines of /var/lib/jenkins/.rvm/log/1429469566_jruby-1.7.11/mvn.log
[2015-04-19 14:52:54] mvn
current path: /var/lib/jenkins/.rvm/src/jruby-1.7.11
PATH=/sbin:/usr/sbin:/bin:/usr/bin:/var/lib/jenkins/.rvm/bin
command(1): mvn
/var/lib/jenkins/.rvm/scripts/functions/utility_logging: line 210: mvn: command not found
jruby-1.7.11 - #gemset created /var/lib/jenkins/.rvm/gems/jruby-1.7.11@resquared
